subprojects/junitJupiterExtensionTest/src/test/java/module-info.java

```diff
@@ -0,0 +1,6 @@
+module mockito.junitJupiterExtensionTest.test {
```

Comment:
Do we also need to specify a `module-info.java` for the main sourceset of this extension? E.g. to denote to users that we require these modules.

Reply:
I'm not certain I understand this question. there is no `main` for this test sub project.

Reply:
oof, I think I realized what you're talking about, and that poses a problem since the extension is an external jar and accesses the internal API. I don't know if there's a way to avoid exporting at least a few of the internal packages for that.

Otherwise these look more reasonable to test this works in the real world.

Reply:
choices, first I want to get it working before making decisions ;) I'm having trouble making it work with some internals stuff, need to figure out the why's.

I started work on this because spring boot blew up on me just for having mockito on my classpath. There was an easy workaround there, but since it looked like PRs were welcome I decided to take a stab ;).

note: might not have to export all of org.mockito.internal, personally I'd rather ship with a subset of that exposed documenting it'll be "hidden" in the next major version.

Reply:
remember a java module is 1 to 1 with a jar, so if an external jar (mockitos or not) depends on any internal package that specific package will have to be exported or perhaps opened. You can export/open to a specific thing though, so perhaps that's enough. Currently though, exporting/opening this would block

```
java.lang.IllegalStateException: Could not initialize plugin: interface org.mockito.plugins.MockMaker (alternate: null)
	at org.mockito/org.mockito.internal.configuration.plugins.PluginLoader$1.invoke(PluginLoader.java:85)
	at jdk.proxy3/jdk.proxy3.$Proxy19.isTypeMockable(Unknown Source)
	at org.mockito/org.mockito.internal.util.MockUtil.typeMockabilityOf(MockUtil.java:78)
	at org.mockito/org.mockito.internal.util.MockCreationValidator.validateType(MockCreationValidator.java:22)
	at org.mockito/org.mockito.internal.creation.MockSettingsImpl.validatedSettings(MockSettingsImpl.java:275)
	at org.mockito/org.mockito.internal.creation.MockSettingsImpl.build(MockSettingsImpl.java:236)
	at org.mockito/org.mockito.internal.MockitoCore.mock(MockitoCore.java:82)
	at org.mockito/org.mockito.Mockito.mock(Mockito.java:2104)
	at org.mockito/org.mockito.internal.configuration.MockAnnotationProcessor.processAnnotationForMock(MockAnnotationProcessor.java:79)
	at org.mockito/org.mockito.internal.configuration.MockAnnotationProcessor.process(MockAnnotationProcessor.java:28)
	at org.mockito/org.mockito.internal.configuration.MockAnnotationProcessor.process(MockAnnotationProcessor.java:25)
	at org.mockito/org.mockito.internal.configuration.IndependentAnnotationEngine.createMockFor(IndependentAnnotationEngine.java:44)
	at org.mockito/org.mockito.internal.configuration.IndependentAnnotationEngine.process(IndependentAnnotationEngine.java:72)
	at org.mockito/org.mockito.internal.configuration.InjectingAnnotationEngine.processIndependentAnnotations(InjectingAnnotationEngine.java:62)
	at org.mockito/org.mockito.internal.configuration.InjectingAnnotationEngine.process(InjectingAnnotationEngine.java:47)
	at org.mockito/org.mockito.MockitoAnnotations.openMocks(MockitoAnnotations.java:81)
	at org.mockito/org.mockito.internal.framework.DefaultMockitoSession.<init>(DefaultMockitoSession.java:43)
	at org.mockito/org.mockito.internal.session.DefaultMockitoSessionBuilder.startMocking(DefaultMockitoSessionBuilder.java:83)
	at org.mockito.junit.jupiter/org.mockito.junit.jupiter.MockitoExtension.beforeEach(MockitoExtension.java:160)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	Suppressed: java.lang.NullPointerException: Cannot invoke "java.util.Set.forEach(java.util.function.Consumer)" because the return value of "org.junit.jupiter.api.extension.ExtensionContext$Store.remove(Object, java.lang.Class)" is null
		at org.mockito.junit.jupiter/org.mockito.junit.jupiter.MockitoExtension.afterEach(MockitoExtension.java:194)
		... 2 more
Caused by: java.lang.IllegalStateException: Internal problem occurred, please report it. Mockito is unable to load the default implementation of class that is a part of Mockito distribution. Failed to load interface org.mockito.plugins.MockMaker
	at org.mockito/org.mockito.internal.configuration.plugins.DefaultMockitoPlugins.create(DefaultMockitoPlugins.java:105)
	at org.mockito/org.mockito.internal.configuration.plugins.DefaultMockitoPlugins.getDefaultPlugin(DefaultMockitoPlugins.java:79)
	at org.mockito/org.mockito.internal.configuration.plugins.PluginLoader.loadPlugin(PluginLoader.java:75)
	at org.mockito/org.mockito.internal.configuration.plugins.PluginLoader.loadPlugin(PluginLoader.java:49)
	at org.mockito/org.mockito.internal.configuration.plugins.PluginRegistry.<init>(PluginRegistry.java:29)
	at org.mockito/org.mockito.internal.configuration.plugins.Plugins.<clinit>(Plugins.java:26)
	at org.mockito/org.mockito.internal.MockitoCore.<clinit>(MockitoCore.java:71)
	at org.mockito/org.mockito.Mockito.<clinit>(Mockito.java:1683)
	at org.mockito.junit.jupiter/org.mockito.junit.jupiter.MockitoExtension.beforeEach(MockitoExtension.java:156)
	... 2 more
```

Reply:
and here I think is a blocking issue. It looks like the code is trying to load `org.junit.jupiter.api.extension.ExtensionContext` reflectively, but junit isn't opening it.

```
module org.junit.jupiter.api {
	requires static transitive org.apiguardian.api;
	requires transitive org.junit.platform.commons;
	requires transitive org.opentest4j;

	exports org.junit.jupiter.api;
	exports org.junit.jupiter.api.condition;
	exports org.junit.jupiter.api.extension;
	exports org.junit.jupiter.api.function;
	exports org.junit.jupiter.api.io;
	exports org.junit.jupiter.api.parallel;

	opens org.junit.jupiter.api.condition to org.junit.platform.commons;
}
```

Reply:
line 180 of `InlineDelegateByteBuddyMockMaker` is throwing `java.lang.ClassNotFoundException: org.mockito.internal.creation.bytebuddy.inject.MockMethodDispatcher` which is currently making no sense to me... will keep digging tomorrow. until I've completely burned out or have something that *could work*.

Reply:
I'm stumped, I'm certain the problem is that `InlineDelegateByteBuddyMockMaker` isn't being loaded because I checked out what was happening without it. Honestly to get full module-info working I need help someone who understands how that static block in `InlineDelegateByteBuddyMockMaker` works. I tried exporting and opening everything to see if that'd fix it. Easier to play clean up on internals once tests pass.

Reply:
btw, I noticed that I accidentally re-ordered some imports (fixed) but I could create a checkstyle check for that in another PR-if y'all wanted.